### DISCUSSION TITLE
Hide app from menu if already in dock

### DISCUSSION THREAD

**Discussion by teamcons** [elementary/applications-menu#586] [Ideas]
Have the app menu not show an app if it is pinned in the dock - In a similar fashion to gnome app drawer and the dash.
Not that there is anything wrong with showing in both - App menu there acts as "here is everything you have on your PC", which is valid.

But showing in app menu may be redundant if in dock - If someone looks in app menu, its not to click something they have under the nose, its to search for something else. And searching for something else means cutting out the noise.

This behavior also keeps the cleanliness and focused way of the desktop : You dont have things in several places. It's here or its there. Simple. No visual noise, only Zuul.

Im happy with both, but if "hide app from menu if already in dock" could be considered, or an option (exposed or not) it would be quite dank.

thank you for your attention !

**Comment by danirabbit** [comment]
Hey thanks for your suggestion! We did a survey around dock behavior and found that making the dock and applications menu mutually exclusive was actually not the expected behavior for most people. You can read more here: https://blog.elementary.io/2021-ui-study-results-dock-multitasking/

**Reply by teamcons** [reply]
Which is fair, but by the time 8 hits, it will make a few years Gnome-shell 40+ have hit, with that behavior as default. And as that shapes what people expect, there may be more coming from there and expecting the same.

in 2021 i was used to deduplicated entries too. There might be more people asking over time, and keeping in sight the ability to switch behavior to be built trivially could reduce headache down the line
